### PR TITLE
Container sanity checks

### LINKED ISSUE #85: Make container sanity checks in container.Put method
It would be nice if node could make sanity checks on container fields at `container.Put` request:
- do not accept containers without owner ID or with invalid owner ID,
- do not accept containers without nonce or with invalid nonce,
- do not accept containers without placement policy,
- do not accept containers with unknown version,
- optional: check container signature (smart-contract won't accept containers with invalid signature anyway, but we can make one more check).

### PR DESCRIPTION
**Pull request by cthulhu-rider**
Closes #85.

### CODE REVIEW

**Comment by alexvanin** [comment]
Don't you mind to add container check to inner ring as well? 

There: pkg/innerring/processors/container/process_container.go
```go
func (cp *Processor) processContainerPut(put *containerEvent.Put) {
	if !cp.activeState.IsActive() {
		cp.log.Info("passive mode, ignore container put")
		return
	}

	// <- convert put.Container() bytes to SDK structure
	// <- if sanity check failed -- log and return

	err := invoke.RegisterContainer(cp.morphClient, cp.containerContract,
```

**Comment by cthulhu-rider** [comment]
> Don't you mind to add container check to inner ring as well?
> 
> There: pkg/innerring/processors/container/process_container.go
> 
> ```go
> func (cp *Processor) processContainerPut(put *containerEvent.Put) {
> 	if !cp.activeState.IsActive() {
> 		cp.log.Info("passive mode, ignore container put")
> 		return
> 	}
> 
> 	// <- convert put.Container() bytes to SDK structure
> 	// <- if sanity check failed -- log and return
> 
> 	err := invoke.RegisterContainer(cp.morphClient, cp.containerContract,
> ```

Done, check pls.

File: pkg/innerring/processors/container/process_container.go [hunk 1]
```diff
@@ -14,10 +19,36 @@ func (cp *Processor) processContainerPut(put *containerEvent.Put) {
 		return
 	}
 
+	cnrData := put.Container()
+
+	// unmarshal container structure
+	// FIXME: temp solution, replace after neofs-api-go#168
+	cnrProto := new(containerGRPC.Container)
+	if err := proto.Unmarshal(cnrData, cnrProto); err != nil {
+		cp.log.Error("could not unmarshal container structure",
```
Comment (alexvanin) [inline pkg/innerring/processors/container/process_container.go:28]: What about logging it at `INFO` or `WARN` level? This is not an internal unhandled error in inner ring. Malformed container is a expected behavior and we process this normally.

File: pkg/innerring/processors/container/process_container.go [hunk 2]
```diff
@@ -14,10 +19,36 @@ func (cp *Processor) processContainerPut(put *containerEvent.Put) {
...
+		return
+	}
+
+	cnr := containerSDK.NewContainerFromV2(
+		v2container.ContainerFromGRPCMessage(cnrProto),
+	)
+
+	// perform format check
+	if err := container.CheckFormat(cnr); err != nil {
+		cp.log.Error("container with incorrect format detected",
```
Comment (alexvanin) [inline pkg/innerring/processors/container/process_container.go:41]: The same log level change proposal.

**Review by alexvanin — APPROVED** [review]

File: pkg/innerring/processors/container/process_container.go (hunk 1 above)
Comment (cthulhu-rider) [reply pkg/innerring/processors/container/process_container.go:28]: I agree, its better to have `INFO` lvl here.

**Pull request merged**